raiden/tests/integration/long_running/test_token_networks.py

```diff
@@ -154,10 +153,26 @@ def test_participant_selection(raiden_network, token_addresses, skip_if_tester):
             assert routes is not None
 
     # create a transfer to the leaving node, so we have a channel to settle
-    sender = raiden_network[-1].raiden
-    receiver = raiden_network[0].raiden
-
+    for app in raiden_network:
+        sender = app.raiden
+        sender_channel = next((
+            channel_state
+            for channel_state in RaidenAPI(sender).get_channel_list(
+                registry_address=registry_address,
+                token_address=token_address,
+            )
+            if channel_state.our_state.contract_balance > 0 and
+            channel_state.partner_state.contract_balance > 0
+        ), None)  # choose a fully funded channel from sender
+        if sender_channel:
+            break
     registry_address = sender.default_registry.address
+
+    receiver = next(
+        app.raiden for app in raiden_network
+        if app.raiden.address == sender_channel.partner_state.address
```

Comment:
what if the loop above is exhausted and not `sender_channel` is set?

Reply:
This never happens, as there **will** be at least 2 nodes with **some** channel which got counter-deposited, per the rules on `ConnectionManager._open_channels`.